src/Library/demos/Picture/main.blp

```diff
@@ -0,0 +1,51 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.StatusPage {
+  title: _("Picture");
+  description: _("Display images at their natural sizes");
+
+  child: Box {
+    orientation: vertical;
+
+    LinkButton {
+      label: _("API Reference");
+      uri: "https://docs.gtk.org/gtk4/class.Picture.html";
+      margin-bottom : 12;
+    }
+
+    Adw.Clamp {
+      maximum-size: 500;
+
+      ListBox {
+        styles ["boxed-list"]
+        margin-top : 12;
+        margin-bottom : 12;
+
+        Adw.ComboRow content_fit_row {
+          title: _("Content Fit");
+          model: StringList {
+            strings ["Fill", "Contain", "Cover", "Scale Down"]
+          };
+        }
+
+        Adw.ActionRow {
+          title: _("Can Shrink");
+          activatable-widget: can_shrink_switch;
+
+          Switch can_shrink_switch {
+            valign: center;
+            active: true;
+          }
+        }
+      }
+    }
+
+     Picture picture {
+      halign:center;
+      valign:center;
+      can-shrink: bind can_shrink_switch.active;
+      content-fit: bind content_fit_row.selected;
```

Comment:
I'm confused, even with `height-request` and `width-request`, `content-fit` doesn't have any effect. I think it might be a bug.

I will keep looking into it. Can you solve the other issues in the meantime?

Reply:
Could you investigate it? You can ask in discourse.gnome.org or the GNOME hackers channel.